lifecycle-operator/webhooks/pod_mutator/handlers/pod_annotation.go

```diff
@@ -0,0 +1,130 @@
+package handlers
+
+import (
+	"context"
+
+	argov1alpha1 "github.com/argoproj/argo-rollouts/pkg/apis/rollouts/v1alpha1"
+	"github.com/go-logr/logr"
+	apicommon "github.com/keptn/lifecycle-toolkit/lifecycle-operator/apis/lifecycle/v1alpha3/common"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+type PodAnnotationHandler struct {
+	Client client.Client
+	Log    logr.Logger
+}
+
+func (p *PodAnnotationHandler) IsAnnotated(ctx context.Context, req *admission.Request, pod *corev1.Pod) bool {
+	podIsAnnotated := isPodAnnotated(pod)
+	if !podIsAnnotated {
+		p.Log.Info("Pod is not annotated, check for parent annotations...")
+		podIsAnnotated = p.copyAnnotationsIfParentAnnotated(ctx, req, pod)
+	}
+	return podIsAnnotated
+}
+
+func (p *PodAnnotationHandler) copyAnnotationsIfParentAnnotated(ctx context.Context, req *admission.Request, pod *corev1.Pod) bool {
```

Comment:
I know that this function is covered in the test of `IsAnnotated`, but let's try to do it differently -> `IsAnnotated` should have only 2 use cases to test -> pod is/is not annotated, the rest should be in the test of the second function -> let's try to split this test into 2, as it will be more visible what is tested where